core-api/src/main/java/com/optimizely/ab/config/parser/ConditionJacksonDeserializer.java

```diff
@@ -0,0 +1,95 @@
+ *    limitations under the License.
+ */
+package com.optimizely.ab.config.parser;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.optimizely.ab.config.audience.AndCondition;
+import com.optimizely.ab.config.audience.AudienceIdCondition;
+import com.optimizely.ab.config.audience.Condition;
+import com.optimizely.ab.config.audience.NotCondition;
+import com.optimizely.ab.config.audience.OrCondition;
+import com.optimizely.ab.config.audience.UserAttribute;
+import com.optimizely.ab.internal.ConditionUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConditionJacksonDeserializer extends JsonDeserializer<Condition> {
+    private ObjectMapper objectMapper;
+
+    public ConditionJacksonDeserializer() {
+        this(new ObjectMapper());
+    }
+
+    ConditionJacksonDeserializer(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public Condition deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        JsonNode node = parser.getCodec().readTree(parser);
+
+        String conditionsJson = node.textValue();
+        JsonNode conditionsTree = objectMapper.readTree(conditionsJson);
+        Condition conditions = ConditionJacksonDeserializer.parseConditions(objectMapper, conditionsTree);
+
+        return conditions;
+    }
+
+    protected static Condition parseConditions(ObjectMapper objectMapper, JsonNode conditionNode) throws JsonProcessingException {
+        List<Condition> conditions = new ArrayList<>();
+        JsonNode opNode = conditionNode.get(0);
+        String operand = opNode.asText();
+
+        for (int i = 1; i < conditionNode.size(); i++) {
+            JsonNode subNode = conditionNode.get(i);
+            if (subNode.isArray()) {
+                conditions.add(ConditionJacksonDeserializer.parseConditions(objectMapper, subNode));
+            } else if (subNode.isTextual()) {
+                conditions.add(objectMapper.treeToValue(subNode, AudienceIdCondition.class));
+            }
+            else if (subNode.isObject()) {
+                conditions.add(objectMapper.treeToValue(subNode, UserAttribute.class));
+            }
+        }
+
+        Condition condition;
+        switch (operand) {
+            case "and":
+                condition = new AndCondition(conditions);
+                break;
+            case "or":
+                condition = new OrCondition(conditions);
+                break;
+            case "not": // this makes two assumptions: operator is "not" and conditions is non-empty...
+                condition = new NotCondition(conditions.get(0));
```

Comment:
Can we handle `conditions.length == 0` a little more gracefully, maybe by constructing a `NullCondition`? Merely because of the precedent that we don't enforce any constraints on `audience.conditions` during initialization.